bot/common.py

```diff
@@ -344,7 +373,8 @@ def process_result(self, result):
             return
         except IsRedirectPage as e:
             wp.debug("{page} is a redirect".format(page=wikipage), layer="")
-            self.fix_redirect(url_gid, e.old, e.new)
+            if self.can_edit:
+                self.fix_redirect(url_gid, e.old, e.new)
```

Comment:
If a redirect gets skipped because it can't edit it at the moment, will it be looked up again later? Or will it still be listed as processed and ignored in the future? (ideally we would probably process it again, but not a huge deal either way, just making sure of how it works)

Reply:
Redirects will be either

1. edited (and their edits_pending attribute will be > 0)
2. just skipped (without updating edits_pending)

but never be marked as processed (in the sense of "has been added to wikidata"). The whole list of `except` blocks here doesn't mark an MBID as processed in the hopes of someone fixing the entry in MB :-)

In case of 1, https://github.com/metabrainz/mb2wikidatabot/blob/6f89fd01357cd2c0dd6286340440c7d702deb701/bot/const.py#L64-L66 will just not try to process the MBID until there are no more open edits.

In case of 2, the MBID will show up in the list of entities-to-process in the next run, to either be edited or skipped again.